src/overmap.cpp

```diff
@@ -601,6 +599,189 @@ void reset_region_settings()
+                    itr != region_settings_map.end(); itr++) {
+                    apply_region_overlay(jo, itr->second);
+                }
+            }
+            else {
+                t_regional_settings_map_itr itr = region_settings_map.find(regionid);
+                if(itr == region_settings_map.end()) {
+                    jo.throw_error("region: " + regionid + " not found in region_settings_map");
+                }
+                else {
+                    apply_region_overlay(jo, itr->second);
+                }
+            }
+        }
+    }
+    else {
+        jo.throw_error("\"regions\" is required and must be an array");
+    }
+}
+
+void apply_region_overlay(JsonObject &jo, regional_settings &region)
+{
+    if (jo.has_member("default_oter")) {
+        jo.read("default_oter", region.default_oter);
+    }
+
+    if (jo.has_object("default_groundcover")) {
+        JsonObject jio = jo.get_object("default_groundcover");
+
+        if (jio.has_member("primary")) {
+            jio.read("primary", region.default_groundcover_str->primary_str);
+        }
+        if (jio.has_member("secondary")) {
+            jio.read("secondary", region.default_groundcover_str->secondary_str);
+        }
+        if (jio.has_member("ratio")) {
+            jio.read("ratio", region.default_groundcover.chance);
+        }
+    }
+
+    if (jo.has_member("num_forests")) {
+        jo.read("num_forests", region.num_forests);
+    }
+    if (jo.has_member("forest_size_min")) {
+        jo.read("forest_size_min", region.forest_size_min);
+    }
+    if (jo.has_member("forest_size_max")) {
+        jo.read("forest_size_max", region.forest_size_max);
+    }
+    if (jo.has_member("house_basement_chance")) {
+        jo.read("house_basement_chance", region.house_basement_chance);
+    }
+    if (jo.has_member("swamp_maxsize")) {
+        jo.read("swamp_maxsize", region.swamp_maxsize);
+    }
+    if (jo.has_member("swamp_river_influence")) {
+        jo.read("swamp_river_influence", region.swamp_river_influence);
+    }
+    if (jo.has_member("swamp_spread_chance")) {
+        jo.read("swamp_spread_chance", region.swamp_spread_chance);
+    }
+
+    if (jo.has_object("field_coverage")) {
+        JsonObject pjo = jo.get_object("field_coverage");
+
+        if (pjo.has_member("percent_coverage")) {
+            double tmpval = 0.0f;
+            pjo.read("percent_coverage", tmpval);
+            region.field_coverage.mpercent_coverage = (int)(tmpval * 10000.0);
+        }
+
+        if (pjo.has_member("default_ter")) {
+            pjo.read("default_ter", region.field_coverage.default_ter_str);
+        }
+
+        if (pjo.has_object("other")) {
+            double tmpval = 0.0f;
+            JsonObject opjo = pjo.get_object("other");
+            std::set<std::string> keys = opjo.get_member_names();
+            for( const auto &key : keys ) {
+                tmpval = 0.0f;
+                if( key != "//" ) {
```

Comment:
@kevingranade Maybe the json loader should *completely* ignore those comment entries? I've seen code in the mapgen area that does not check this and will choke upon a comment. The json loader already accepts several members of this name in one object, which is invalid according to the json specification, so it would not be that much worse.